From 1d9aa7452aa91133a99c0aa572751f389a115d43 Mon Sep 17 00:00:00 2001
From: Uri Mashiach <uri.mashiach@compulab.co.il>
Date: Sun, 17 Jan 2021 18:37:55 +0200
Subject: [PATCH] Add build targets for CL-SOM-iMX8X

	make SOC=iMX8QX cl_som_imx8x
	make SOC=iMX8QX imx-boot-cl-som-imx8x-sf.bin
	make SOC=iMX8QX imx-boot-cl-som-imx8x-sd.bin

Signed-off-by: Igor Derzhavets <igor.derzhavets@compulab.co.il>
---
 iMX8QX/soc.mak | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/iMX8QX/soc.mak b/iMX8QX/soc.mak
index 1ec1ec0..261c34a 100755
--- a/iMX8QX/soc.mak
+++ b/iMX8QX/soc.mak
@@ -128,6 +128,25 @@ flash_linux_m4_xip: $(MKIMG) $(AHAB_IMG) scfw_tcm.bin u-boot-atf-container.img m
 flash_regression_linux_m4: $(MKIMG) $(AHAB_IMG) scfw_tcm.bin u-boot-atf.bin m4_image.bin
 	./$(MKIMG) -soc QX -rev B0 -append $(AHAB_IMG) -c -flags 0x00200000 -scfw scfw_tcm.bin -ap u-boot-atf.bin a35 0x80000000 -p3 -m4 m4_image.bin 0 0x34FE0000 -out flash.bin
 
+imx-boot-cl-som-imx8x-sd.bin: $(MKIMG) $(AHAB_IMG) scfw_tcm.bin u-boot-spl.bin u-boot-atf-container.img
+	./$(MKIMG) -soc QX -rev B0 -dcd skip -append $(AHAB_IMG) -c -scfw scfw_tcm.bin -ap u-boot-spl.bin a35 0x00100000 -out flash.bin
+	@flashbin_size=`wc -c flash.bin | awk '{print $$1}'`; \
+                   pad_cnt=$$(((flashbin_size + 0x400 - 1) / 0x400)); \
+                   echo "append u-boot-atf-container.img at $$pad_cnt KB"; \
+                   dd if=u-boot-atf-container.img of=flash.bin bs=1K seek=$$pad_cnt;
+	cp flash.bin imx-boot-cl-som-imx8x-sd.bin
+
+imx-boot-cl-som-imx8x-sf.bin: $(MKIMG) $(AHAB_IMG) scfw_tcm.bin u-boot-spl.bin u-boot-atf-container.img
+	./$(MKIMG) -soc QX -rev B0 -dcd skip -dev flexspi -append $(AHAB_IMG) -c -scfw scfw_tcm.bin -ap u-boot-spl.bin a35 0x00100000 -out flash.bin
+	@flashbin_size=`wc -c flash.bin | awk '{print $$1}'`; \
+                   pad_cnt=$$(((flashbin_size + 0x400 - 1) / 0x400)); \
+                   echo "append u-boot-atf-container.img at $$pad_cnt KB"; \
+                   dd if=u-boot-atf-container.img of=flash.bin bs=1K seek=$$pad_cnt;
+	./$(QSPI_PACKER) $(QSPI_HEADER)
+	cp flash.bin imx-boot-cl-som-imx8x-sf.bin
+
+cl_som_imx8x: imx-boot-cl-som-imx8x-sd.bin imx-boot-cl-som-imx8x-sf.bin
+
 flash_regression_linux_m4_ddr: $(MKIMG) $(AHAB_IMG) scfw_tcm.bin u-boot-atf.bin m4_image.bin
 	./$(MKIMG) -soc QX -rev B0 -append $(AHAB_IMG) -c -flags 0x00200000 -scfw scfw_tcm.bin -ap u-boot-atf.bin a35 0x80000000 -p3 -m4 m4_image.bin 0 0x88000000 -out flash.bin
 
-- 
2.11.0

